src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -162,13 +164,28 @@ public boolean isOnStreetAllTheWay() {
     return isStreetOnly();
   }
 
+  /**
+   * Returns true if this itinerary has only flex and walking legs.
+   */
+  public boolean isFlexAndWalkOnly() {
+    return legs.stream().allMatch(l -> l.isFlexibleTrip() || l.isWalkingLeg());
+  }
```

Comment:
Direct flex results is what we are trying to check for here, correct. I'm not sure if the source (DirectFlexRouter) is relevant. Since I have no strong opinion I will change it.